### PR TITLE
Feature: Implement posts filtering

### PR DESCRIPTION
**Pull request by hdn7**

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -1,8 +1,51 @@
...
+  const onKeywordsChange = e => {
+    const keywords = e.target.value;
+    setFilteredPosts(
+      posts
+        .map(post => {
+          const postData = [post.title, post.description, post.content].join(' ');
+          post.matchedLength = lcs(postData, keywords);
+          return post;
+        })
+        .sort((post1, post2) => (post1.matchedLength > post2.matchedLength ? '-1' : '1'))
```
Comment (hdn7) [inline pages/index.js:42]: This causes some ESLint problems right now. Can you think of a better way to do this ?

File: pages/index.js [hunk 2]
```diff
@@ -1,8 +1,51 @@
...
+const useFilter = posts => {
+  const [filteredPosts, setFilteredPosts] = useState(posts);
+
+  const onKeywordsChange = e => {
+    const keywords = e.target.value;
+    setFilteredPosts(
+      posts
+        .map(post => {
+          const postData = [post.title, post.description, post.content].join(' ');
+          post.matchedLength = lcs(postData, keywords);
```
Comment (hdn7) [inline pages/index.js:39]: This breaks ESLint's no-param-reassign rule. Can you think of a better way to do this ?

File: pages/index.js (hunk 1 above)
Comment (hdn7) [inline pages/index.js:42]: Line 39 breaks ESLint's no-param-reassign rule. Can you think of a better way to do this ?

Comment (alexfertel) [reply pages/index.js:42]: yeah, you shouldn't ever try to modify params. `map` is a *transformation* of each element of the list. In this case, transforming means that you compute a new list. You are not supposed to modify the list you are mapping over, that's a source of bugs...state should be immutable when possible (cause we're in javascript :grin:).

```suggestion
        .map(post => {
          const postData = [post.title, post.description, post.content].join(' ');
          
          return {
              ...post,
              matchedLength: lcs(postData, keywords)
          };
        })
        .sort((post1, post2) => (post1.matchedLength > post2.matchedLength ? '-1' : '1'))
```

**Pull request merged**